gax/src/main/java/com/google/api/gax/rpc/Watchdog.java

```diff
@@ -138,12 +140,18 @@ public void shutdown() {
 
   @Override
   public boolean isShutdown() {
-    return executor.isShutdown();
+    if (shouldShutdownExecutor) {
+      return executor.isShutdown();
+    }
+    return true;
```

Comment:
Should this return whether the future is done instead of returning true in the case where we aren't shutting down the executor?